File: corpus/10_group.py
from collections import defaultdict
sales = [{"city":"NY","amt":50}, {"city":"LA","amt":30}, {"city":"NY","amt":20}]
g = defaultdict(int)
for s in sales: g[s["city"]] += s["amt"]
for k, v in g.items(): print(k, v)

File: corpus/10_group.rs
use std::collections::HashMap;

fn main() {
    let sales = [("NY", 50), ("LA", 30), ("NY", 20)];
    let mut g: HashMap<&str, i64> = HashMap::new();
    for (city, amt) in sales {
        *g.entry(city).or_insert(0) += amt;
    }
    for (k, v) in &g {
        println!("{k} {v}");
    }
}
